sources/parsers/parsers.ml

```diff
@@ -132,10 +155,24 @@ let parse_input_file file =
   in
   try
     (if Options.infer_output_format () then
-       match ext with
-       | ".smt2" | ".psmt2" -> Options.set_output_format OSmtlib
-       | ".why" | ".mlw" | ".ae" -> Options.set_output_format ONative
-       | _ -> Format.eprintf "Warning: This extension is not supported@."
+       let ext =
+         try Str.string_after ext 1
+         with Invalid_argument _ -> ext
+       in
+       match Options.match_format ext with
+       | "native" -> Options.set_output_format ONative
+       | "smtlib" -> Options.set_output_format OSmtlib
+       | "why3" -> Options.set_output_format OWhy3
+       (* | "szs" -> Options.set_output_format OSZS *)
+       | _ ->
+         if Options.infer_input_format () then
+           Format.eprintf "Warning: This extension is not supported@."
+         else if Options.input_native () then
+           Options.set_output_format ONative
+         else if Options.input_smtlib () then
+           Options.set_output_format OSmtlib
+         else if Options.input_why3 () then
+           Options.set_output_format OWhy3
```

Comment:
This nesting of if/then/elses seems really strange, why not pattern_match on the input_format since it exists somewhere ?